pkgs/tools/misc/graylog/graylog.nix

```diff
@@ -0,0 +1,47 @@
+{ lib, stdenv, fetchurl, makeWrapper, openjdk11_headless, openjdk17_headless, systemd, nixosTests}:
+
+{ version, sha256, maintainers, license }:
+
+let
+  envLibPath = lib.makeLibraryPath [
+    systemd
+  ];
+in stdenv.mkDerivation rec {
+  pname = "graylog";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://packages.graylog2.org/releases/graylog/graylog-${version}.tgz";
+    inherit sha256;
+  };
+
+  dontBuild = true;
+  dontStrip = true;
+
+  nativeBuildInputs = [ makeWrapper ];
+  makeWrapperArgs = [
+    "--set-default" "JAVA_HOME" "${if (lib.versionAtLeast version "5.0") then openjdk17_headless else openjdk11_headless}"
+    "--prefix LD_LIBRARY_PATH : ${envLibPath}"
+  ];
+
+  passthru.tests = { inherit (nixosTests) graylog; };
+
+  installPhase = ''
+    mkdir -p $out
+    cp -r {graylog.jar,bin,plugin} $out
```

Comment:
We should still copy the files into a FHS like structure for packages and not just move them to $out. Probably $out/share/graylog or similar would be a better place.